src/components/template-selector-control.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * External dependencies
+ */
+import { isEmpty } from 'lodash';
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { withInstanceId } from '@wordpress/compose';
+import { BaseControl } from '@wordpress/components';
+import { createElement } from '@wordpress/element';
+
+
+
+function TemplateSelectorControl( { label, className, selected, help, instanceId, onChange, templates = [] } ) {
```

Comment:
Should this be "selectorControl" or "radioControl"?

Reply:
So (as you know) Gutenberg has `RadioControl` which is what this is based on. However doesn't afford the flexibility we need our needs are much more specific to Templates. Therefore I opted to give it a very specific name. 

I did call it `TemplateRadioControl` but felt that didn't convey the purpose really because it is ultimately a "selector".